sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -0,0 +1,208 @@
+
+            try
+            {
+                explodedGraph.Walk();
+            }
+            finally
+            {
+                objectDisposedCheck.ObjectDisposed -= memberAccessedHandler;
+            }
+        }
+
+        internal class ObjectDisposedEventArgs : EventArgs
+        {
+            public string Name { get; }
+            public Location Location { get; }
+
+            public ObjectDisposedEventArgs(string name, Location location)
+            {
+                Name = name;
+                Location = location;
+            }
+        }
+
+        internal sealed class ObjectDisposedPointerCheck : ExplodedGraphCheck
+        {
+            public event EventHandler<ObjectDisposedEventArgs> ObjectDisposed;
+
+            public ObjectDisposedPointerCheck(ExplodedGraph explodedGraph)
+                : base(explodedGraph)
+            {
+            }
+
+            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                var instruction = programPoint.Block.Instructions[programPoint.Offset] as InvocationExpressionSyntax;
+
+                return instruction == null
+                    ? programState
+                    : VisitInvocationExpression(instruction, programState);
+            }
+
+            public override ProgramState PreProcessUsingStatement(ProgramPoint programPoint, ProgramState programState)
+            {
+                var newProgramState = programState;
+
+                var usingFinalizer = (UsingFinalizerBlock)programPoint.Block;
+
+                var disposables = usingFinalizer.Disposables
+                    .Select(disposable =>
+                    new
+                    {
+                        Disposable = disposable,
+                        Symbol = semanticModel.GetDeclaredSymbol(disposable)
+                            ?? semanticModel.GetSymbolInfo(disposable).Symbol
+                    });
+
+                foreach (var nodeWithSymbol in disposables)
+                {
+                    newProgramState = ProcessDisposableSymbol(newProgramState, nodeWithSymbol.Disposable, nodeWithSymbol.Symbol);
+                }
+
+                newProgramState = ProcessStreamDisposingTypes(newProgramState, (SyntaxNode)usingFinalizer.UsingStatement.Expression ?? usingFinalizer.UsingStatement.Declaration);
+
+                return newProgramState;
+            }
+
+            private ProgramState ProcessStreamDisposingTypes(ProgramState programState, SyntaxNode usingExpression)
+            {
+                var newProgramState = programState;
+
+                var arguments = usingExpression.DescendantNodes()
+                    .OfType<ObjectCreationExpressionSyntax>()
+                    .Where(this.IsStreamDisposingType)
+                    .Select(FirstArgumentOrDefault)
+                    .WhereNotNull();
+
+                foreach (var argument in arguments)

```

Comment:
What about another `Select` to have syntax + symbol and then `ToList().ForEach()`?